src/Core/OEGlobalsBag.php

```diff
@@ -17,16 +22,29 @@ public static function getInstance()
         return self::$instance;
     }
 
+    /**
+     * Enables or disables compatibility mode for OEGlobalsBag which will populate the $GLOBALS array for backwards
+     * compatibility with legacy code.
+     * @param bool $enabled
+     * @return void
+     */
+    public static function setCompatabilityMode(bool $enabled): void
+    {
+        $instance = self::getInstance();
+        $instance->compatHolder->enabled = $enabled;
+    }
+
     public function __construct(array $parameters = [], private readonly bool $compatabilityMode = false)
     {
         parent::__construct($parameters);
+        $this->compatHolder = (object) ['enabled' => $this->compatabilityMode];
     }
 
     public function set(string $key, mixed $value): void
     {
         $this->parameters[$key] = $value;
-        if ($this->compatabilityMode) {
-            // In compatibility mode, also set the value in the global $_GLOBALS array
+        if ($this->compatHolder->enabled) {
+            // In compatibility mode, also set the value in the global $GLOBALS array
```

Comment:
@kojiromike I believe this may be a duplicate solution to https://github.com/openemr/openemr/pull/9410/files#diff-f85e48dbc52b2f7b6f6e19c006a460e7997e2c4e20df559570a9fb6cebbf4f10